src/domain/dtos/appointment.dto.ts

```diff
@@ -0,0 +1,69 @@
+import { CustomError } from "../errors/customError";
+import { AppointmentStatus } from "../types/appointmentStatus.type";
+
+
+export class AppointmentDTO {
+  
+
+  private constructor(
+    public patientId: string,
+    public doctorId: string,
+    public dateTime: Date,
+    public status: AppointmentStatus,
+    public reason?: string,
+    public notes?: string,
+    public id?: string,
+  ) {}
+
+  public static validate(data: {[key: string]: any}): [string | null, AppointmentDTO | null] {
+
+    if (data.patientId !== undefined && data.patientId !== null) {
+      if (typeof data.patientId !== 'string') {
+        return ['Patient ID must be a string', null];
+      }
+    }
```

Comment:
```suggestion
    if (data.patientId !== undefined && data.patientId !== null) {
      if (typeof data.patientId !== 'string') {
        return ['Patient ID must be a string', null];
      }
    }
```
Esta misma secuencia se repite para todos los casos, en estos escenarios, sí es buena idea pensar en una utilidad para poder reutilizar desde distintos dtos, porque sino, lo más probable es que termines teniendo esta misma secuencia de ifs, con el mismo método validate, para distintos dtos... cuando podrías tener un dto normal, con un método genérico que resuelva para todos este comportamiento... Incluso habría que validar, si estos datos son requeridos, no es necesario validarlos, si es required, te aseguras que no será undefined, ni null, y que además será el tipo que corresponde, sino, rompe el contrato

Reply:
Con el siguiente bloque se resuelve tu inquietud:
https://github.com/candiath/clinex/blob/98aaacd87b7a956fe2d50e8c004ea921fa4c87e7/src/domain/dtos/appointment.dto.ts#L22-L27